applications/StructuralMechanicsApplication/custom_advanced_constitutive/generic_small_strain_high_cycle_fatigue_law.h

```diff
@@ -173,34 +180,47 @@ class KRATOS_API(STRUCTURAL_MECHANICS_APPLICATION) GenericSmallStrainHighCycleFa
     bool Has(const Variable<int>& rThisVariable) override;
 
     /**
-     * @brief Sets the value of a specified variable (double)
-     * @param rVariable the variable to be returned
-     * @param rValue new value of the specified variable
+     * @brief Sets the value of a specified variable (bool)
+     * @param rThisVariable the variable to be returned
+     * @param Value new value of the specified variable
      * @param rCurrentProcessInfo the process info
      */
     void SetValue(
-        const Variable<double>& rThisVariable,
-        const double& rValue,
+        const Variable<bool>& rThisVariable, 
+        const bool& Value, 
         const ProcessInfo& rCurrentProcessInfo) override;
 
     /**
      * @brief Sets the value of a specified variable (integer)
+     * @param rThisVariable the variable to be returned
+     * @param rValue new value of the specified variable
+     * @param rCurrentProcessInfo the process info
+     */
+    void SetValue(
+        const Variable<int>& rThisVariable, 
+        const int& rValue,
+        const ProcessInfo& rCurrentProcessInfo) override;
+
+    /**
+     * @brief Sets the value of a specified variable (double)
      * @param rVariable the variable to be returned
      * @param rValue new value of the specified variable
+     * @param rCurrentProcessInfo the process info
      */
     void SetValue(
-        const Variable<int>& rThisVariable,
-        const int& rValue);
+        const Variable<double>& rThisVariable, 
+        const double& rValue,
+        const ProcessInfo& rCurrentProcessInfo) override;
 
     /**
-     * @brief Returns the value of a specified variable (double)
+     * returns the value of a specified variable
      * @param rThisVariable the variable to be returned
      * @param rValue a reference to the returned value
-     * @return rValue output: the value of the specified variable
+     * @param rValue output: the value of the specified variable
      */
-    double& GetValue(
-        const Variable<double>& rThisVariable,
-        double& rValue) override;
+    bool& GetValue(
+        const Variable<bool>& rThisVariable, 
+        bool& rValue) override;
```

Comment:
now I understand why there are the clang warnings, the one in the baseclass also gets a ProcessInfo

@AlejandroCornejo how can this work with override? => why not using the baseclass functions?

Reply:
oh this can explain several things. It is weird that it compiles wqith the override right??

Reply:
yes exactly
maybe you have some intermediate class that defines the new functions?

Reply:
This is from the constitutive_law.h methods:

`    /**
     * @brief Returns the value of a specified variable (boolean)
     * @param rThisVariable the variable to be returned
     * @param rValue a reference to the returned value
     * @return rValue output: the value of the specified variable
     */
    virtual bool& GetValue(const Variable<bool>& rThisVariable, bool& rValue);`

And there is no ProcessInfo. I had this problem with the SetValue which requires ProcessInfo, but is not the case of theGetValue

Reply:
I see, thx for clarifying

this is a bit messed up I guess but not sure we can do much abt it...

then I guess I will take a look at the clang warnings myself at some point

any comment @RiccardoRossi